### ISSUE TITLE
Improve unit test coverage in API module

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sshyran/genxdm#25]
```
In the API module, all concrete classes (including, in fact, abstract classes 
as well, which can be tested by minimal extension) need to have unit tests that 
verify conformance with their contracts.

It appears that the current test coverage is incomplete.
```

Original issue reported on code.google.com by `aale...@gmail.com` on 20 Oct 2010 at 4:32
- Blocked on: #39

**Comment by GoogleCodeExporter** [comment]
```
The list of classes follows:

org.genxdm.compat.DomCompatibility

org.genxdm.exceptions.*, except AssertMessage and SpillagePolicy

org.genxdm.names.NameSource

org.genxdm.Resolved

org.genxdm.xs.constraint.NamespaceConstraint
org.genxdm.xs.constraint.ValueConstraint

org.genxdm.xs.exceptions.*, except SchemaExceptionThrower and 
SchemaExceptionHandler

org.genxdm.xs.SchemaLoadOptions

org.genxdm.xs.types.UberTypeArgumentOrSelfItera{tor,ble}

Some of these are not easily tested (not public, for instance, or 
parameterized).

Still, we ought to either cover them in unit tests, or explain why we don't.

There are also some enums that might have sufficient behavior to be worth 
testing--in so far as something has a contract, it would be nice to have it 
tested just as soon as it is concrete.
```

Original comment by `aale...@gmail.com` on 21 Oct 2010 at 8:02

**Comment by GoogleCodeExporter** [comment]
```
In process.
```

Original comment by `aale...@gmail.com` on 11 Nov 2010 at 6:48
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
There are a number of changes as of r87.  We've switched to junit 4.  Issue 39 
blocks this one, because issue 39 requires us to figure out the tangle for 
Quantifier (in typed.types) and KeeneQuantifier (in xs.enums).

What remains to be done is noted in "COVERAGE" in the apitest package.  The 
largest remaining area is the schema model stuff; there are a number of things 
there.

What's there, and live (total of 24 tests) seems to *mostly* follow our new 
rules: clear contract-based testing (not necessarily true for the NativeType 
test case, which contributes a lot of tests, but it's part of schema model and 
will be reviewed as that is reviewed and better understood).
```

Original comment by `aale...@gmail.com` on 15 Dec 2010 at 6:45

**Comment by GoogleCodeExporter** [comment]
```
All testing should be complete for 1.0
```

Original comment by `aale...@gmail.com` on 26 Jul 2012 at 7:44
- Added labels: **Milestone-Release1.0**
- Now blocked on: #39

**Comment by GoogleCodeExporter** [comment]
```
Defer
```

Original comment by `aale...@gmail.com` on 24 Oct 2013 at 5:25
- Added labels: **Milestone-Release1.1**
- Removed labels: **Milestone-Release1.0**
